pkgs/applications/misc/hello/default.nix

```diff
@@ -16,8 +18,16 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
-  passthru.tests.version =
-    testVersion { package = hello; };
+  passthru.tests = {
+    version =
+      testVersion { package = hello; };
+
+    invariant-under-noXlibs =
+      testEqualDerivation
+        "hello must not be rebuilt when environment.noXlibs is set."
+        hello
+        (nixos { environment.noXlibs = true; })._module.args.pkgs.hello;
```

Comment:
Are we 'allowed' to refer to `_module`?

Reply:
Preferably not. `nixos` exposes `pkgs` directly, so I'll use that. (`evalModules` doesn't)

Reply:
Done. Thanks!